tests/ceph_installer/test_ansible.py

```diff
@@ -27,11 +27,26 @@ def run(ceph_cluster, **kw):
     mixed_lvm_configs = config.get('is_mixed_lvm_configs', None)
     device_to_add = config.get('device', None)
     config['ansi_config']['public_network'] = get_public_network(ceph_nodes[0])
-
     ceph_cluster.ansible_config = config['ansi_config']
     ceph_cluster.custom_config = test_data.get('custom-config')
     ceph_cluster.custom_config_file = test_data.get('custom-config-file')
 
+    if config['ansi_config']['rgw_multisite'] and config['ansi_config']['rgw_zonesecondary'] is True:
+        rgw = []
+        ceph_cluster_dict = kw.get('ceph_cluster_dict')
+        for cluster_name in ceph_cluster_dict.keys():
+            primary_node = 'ceph-rgw1'
+            if cluster_name == primary_node:
+                temp_node = ceph_cluster_dict[cluster_name]
+                role = 'rgw'
+                for rnode in temp_node:
+                    if rnode.role == role:
+                        rgw.append(rnode)
```

Comment:
looks like this block of code is for finding out rgw node from the primary cluster. 
couldn't this be written in a short form like this: 

```
primary_node = 'ceph-rgw1'
primary_cluster_rgw = ceph_cluster_dict.get(primary_node).role.get('rgw')
```
then you can find the `ip_address` from `primary_cluster_rgw` . test it out once if you incorporate it.

Reply:
I think you could even use `lambda` for comparison. the above code snippet which I added is just by looking at it. I may be wrong here in unpacking. but this can definitely be reduced something similar above. check it out once